packages/neon_http_client/lib/src/client.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:cookie_store/cookie_store.dart';
+import 'package:http/http.dart' as http;
+import 'package:neon_http_client/src/interceptors/interceptors.dart';
+import 'package:universal_io/io.dart';
+
+/// An exception caused by an error in a [NeonClient].
+sealed class NeonClientException extends http.ClientException {
+  NeonClientException(super.message, [super.uri]);
+}
+
+/// An exception caused by a [HttpInterceptor].
+final class InterceptionFailure extends NeonClientException {
+  /// Creates a new interceptor failure exception.
+  InterceptionFailure(super.message, [super.uri]);
+}
+
+/// A client configuring the clients for all Nextcloud APIs.
+///
+/// To access the APIs of a particular app import the extensions through `package:nextcloud/{id}.dart`.
+final class NeonClient with http.BaseClient {
```

Comment:
```suggestion
final class NeonHttpClient with http.BaseClient {
```
Otherwise the name is a bit ambigous

Reply:
I also renamed the file to match the class name.
We have way to many `client.dart` files